artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/JsonUtil.java

```diff
@@ -173,7 +172,7 @@ public static void addToObject(final String key, final Object param, final JsonO
          }
          jsonObjectBuilder.add(key, objectArrayBuilder);
       } else {
-         throw ActiveMQClientMessageBundle.BUNDLE.invalidManagementParam(param.getClass().getName());
```

Comment:
You missed one thing here...


add an instance Number.. and add a double number...


look at my original proposal.

Reply:
```
 public static void addToObject(final String key, final Object param, final JsonObjectBuilder jsonObjectBuilder) {
      if (param instanceof Integer) {
         jsonObjectBuilder.add(key, (Integer) param);
      } else if (param instanceof Long) {
         jsonObjectBuilder.add(key, (Long) param);
      } else if (param instanceof Double) {
         jsonObjectBuilder.add(key, (Double) param);
      } else if (param instanceof String) {
         jsonObjectBuilder.add(key, (String) param);
      } else if (param instanceof Boolean) {
         jsonObjectBuilder.add(key, (Boolean) param);
      } else if (param instanceof Map) {
         JsonObject mapObject = toJsonObject((Map<String, Object>) param);
         jsonObjectBuilder.add(key, mapObject);
      } else if (param instanceof Short) {
         jsonObjectBuilder.add(key, (Short) param);
      } else if (param instanceof Byte) {
         jsonObjectBuilder.add(key, ((Byte) param).shortValue());
      } else if (param instanceof SimpleString) {
         jsonObjectBuilder.add(key, param.toString());
      } else if (param == null) {
         jsonObjectBuilder.addNull(key);
      } else if (param instanceof byte[]) {
         JsonArrayBuilder byteArrayObject = toJsonArrayBuilder((byte[]) param);
         jsonObjectBuilder.add(key, byteArrayObject);
      } else if (param instanceof Object[]) {
         final JsonArrayBuilder objectArrayBuilder = JsonLoader.createArrayBuilder();
         for (Object parameter : (Object[])param) {
            addToArray(parameter, objectArrayBuilder);
         }
         jsonObjectBuilder.add(key, objectArrayBuilder);
      } else if (param instanceof Number) {
         jsonObjectBuilder.add(key, ((Number)param).doubleValue());
      } else {
         jsonObjectBuilder.add(key, param.toString());
      }
   }
```

the instance Number is important

Reply:
don't necessarily copy my version though. what I meant was the param instanceof Number..

you probably should do the same for the Object[] ? just a suggestion though.